### PR TITLE
Added FlxSound

### PR DESCRIPTION
**Pull request by NaxeCode**
Feel free to replace the current sound if need be.
(Points if you know where the current one is from)

### CODE REVIEW

File: _proofs/sound/flxsound.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: FlxSound
+concept: Sound
+complexity: 1
+tags: [sound]
+order: 1
+---
+{% api flixel.FlxG %} provides a way of playing sounds, should you not feel like initiating a FlxSound instance.
```
Comment (Gama11) [inline _proofs/sound/flxsound.md]: `FlxSound` should probably be an API link too, like `FlxG`. :)

File: demos/sound/flxsound/source/PlayState.hx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import flixel.math.FlxMath;
+
+class PlayState extends FlxState
+{
+	override public function create():Void
+	{
+		bgColor = 0;
+		super.create();
+
+		var soundBox:FlxButton = new FlxButton(0, 0, "Play Sound", playSound);
```
Comment (Gama11) [inline demos/sound/flxsound/source/PlayState.hx]: Code-style wise, the first type hint here is redundant - it's better style to rely on type inference:

```haxe
var soundBox = new FlxButton(0, 0, "Play Sound", playSound);
```

Comment (Gama11) [inline demos/sound/flxsound/source/PlayState.hx]: Similarly, I would probably declare `playSound()` as an inline / anonymous function myself.

File: _proofs/sound/flxsound.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+```haxe
+// Play a sound
+FlxG.sound.play();
```
Comment (Gama11) [inline _proofs/sound/flxsound.md]: Might want to have some dummy argument in this example call, like `FlxG.sound.play(pathToSoundAsset);`.

**Comment by Gama11** [comment]
Not sure what's wrong with Travis, but it was able to build your new demo successfully it looks like. I think Travis may just not be set up correctly to build pull requests (@SeiferTim?).

**Comment by SeiferTim** [comment]
Okay, it failed because it tried to upload to my server and didn't know the password... hmm...

How does haxeflixel.com do it? it doesn't upload to the live site for a PR, right? It goes to preview.haxeflixel.com?

**Comment by Gama11** [comment]
haxeflixel.com doesn't upload PRs at all - that would be quite dangerous, anybody could inject malicious code. In fact, I've disabled building or PRs in the Travis config for haxeflixel.com because they would always only error anyway.

However, I wouldn't do that in this case, it still seems useful to know whether the examples all build after a PR. Just make it skip the upload step for PRs I guess?

**Comment by SeiferTim** [comment]
@NaxeCode We want to disable uploading for PRs, so, in `.travis.yml`, near the bottom of the file, there's an `if [[ "$TRAVIS_REPO_SLUG" == "HaxeFlixel/haxeflixel-mechanics" ]]; then` line, change it to be:

`if [[ "$TRAVIS_REPO_SLUG" == "HaxeFlixel/haxeflixel-mechanics" && "$TRAVIS_PULL_REQUEST" == false ]]; then`

And then re-commit.

**Comment by Gama11** [comment]
Does it really matter for this PR? We could just change it on master, no big deal if Travis remains broken for this PR (and is then OK in the future).

**Pull request merged**